dandiapi/zarr/checksums.py

```diff
@@ -21,6 +22,18 @@
     ZarrJSONChecksumSerializer,
 )
 
+ZARR_CHECKSUM_REGEX = r'[0-9a-f]+-(\d+)--(\d+)'
+
+
+def parse_checksum_string(checksum: str) -> tuple[int, int]:
+    """Return a tuple of (file count, total size)."""
+    match = re.match(ZARR_CHECKSUM_REGEX, checksum)
+    if match is None:
+        raise Exception('Invalid zarr checksum provided.')
+
+    count, size = match.groups()
+    return (int(count), int(size))
```

Comment:
This is subjective, but I wonder if we should just have two `@property`s on the Zarr model, `file_count` and `total_size`, instead of a function that we call exactly once. I think there's a lot of places we use properties incorrectly on our models, but these two values seem like a perfect use case as they are derived purely from the `checksum` field (usually I use this rationale https://github.com/HackSoftware/Django-Styleguide#properties for determining when something should be a model property)

Reply:
Very good point, I agree. I'll merge this PR as is and open another one to address this.